File: insertion_sort.py
def insertion_sort(array):
    size = len(array)
    for i in range(1, size):
        key = array[i]
        j = i - 1
        while j >= 0 and array[j] > key:
            array[j + 1] = array[j]
            j = j - 1
        array[j + 1] = key
    return array

print('Demonstration of Insertion Sort')
print('Enter the size of the array = ')
n = int(input())
array = []
print(f"Enter {n} integers = ")
for i in range(0, n):
    array.append(int(input()))
print('Sorted array')
result = insertion_sort(array)
for element in result:
    print(f"{element}, ", end="")

File: insertion_sort.c
#include<stdio.h>
#include<stdlib.h>
void insertion_sort(int* array, int length) {
	for(int i = 1; i < length; i++) {
		int key = array[i];
		int j = i - 1;
		while(j >= 0 && array[j] > key){
			array[j + 1] = array[j];
			j--;
		}
		array[j + 1] = key;
	}
}

int main() {
	printf("Demonstration of Insertion Sort\n");
	printf("Enter the size of the array = \n");
	int length;
	scanf("%d", &length);
	int* array = (int*) calloc(length, sizeof(int));
	printf("Enter %d integers: \n", length);
	for(int i = 0; i < length; i++)
		scanf("%d", &array[i]);
	insertion_sort(array, length);
	printf("Printing sorted array: ");
	for(int i = 0; i < length; i++)
		printf("%d ", array[i]);
	printf("\n");
	return 0;
}
